crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -276,30 +279,35 @@ fn put(
 
 #[put("")]
 async fn put_handler(
+    state: Data<AppState>,
     req: Json<PutReq>,
     mut db_conn: DbConnection,
     user: User,
 ) -> superposition::Result<Json<PutResp>> {
-    put(req, &mut db_conn, false, &user)
-        .map(|resp| Json(resp))
-        .map_err(|err: superposition::AppError| {
-            log::info!("context put failed with error: {:?}", err);
-            err
-        })
+    let version_id = generate_snowflake_id(&state)?.to_string();
+    db_conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
+        let put_response = put(req, transaction_conn, &user)
+            .map(|resp| Json(resp))
+            .map_err(|err: superposition::AppError| {
+                log::info!("context put failed with error: {:?}", err);
+                err
+            })?;
+        add_config_version(version_id, transaction_conn)?;
```

Comment:
Just a query, should we fail the context/default_config generation, if the updation of `config_version` fails?

Reply:
yes